caravel/assets/javascripts/SqlLab/components/QuerySearchBar.jsx

```diff
@@ -0,0 +1,147 @@
+const $ = window.$ = require('jquery');
+import React from 'react';
+
+import { Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Select from 'react-select';
+import * as Actions from '../actions';
+
+class QuerySearchBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      databaseLoading: false,
```

Comment:
you could just check if the list of the databases is empty or not rather than keeping it in separate var.